src/SDKs/BotService/BotService.Tests/Helpers/BotServiceManagementTestUtilities.cs

```diff
@@ -0,0 +1,200 @@
+                Tags = DefaultTags,
+                Properties = new BotProperties()
+                {
+                    Description = "Default bot description",
+                    DisplayName = "Default bot display name",
+                    Endpoint = "https://mybot.coffee",
+                    MsaAppId = Guid.NewGuid().ToString(),
+                    MsaAppPassword = Guid.NewGuid().ToString()
+                }
+            };
+        }
+
+        public static string CreateResourceGroup(ResourceManagementClient resourcesClient)
+        {
+            const string testPrefix = "res";
+            var rgname = TestUtilities.GenerateName(testPrefix);
+
+            if (!IsTestTenant)
+            {
+                var resourceGroup = resourcesClient.ResourceGroups.CreateOrUpdate(
+                    rgname,
+                    new ResourceGroup
+                    {
+                        Location = "centralus"
+                    });
+            }
+
+            return rgname;
+        }
+
+        public static Bot CreateAndValidateBot(AzureBotServiceClient botServiceMgmtClient, string rgname, string kind = null)
+        {
+            string botid = TestUtilities.GenerateName("botservice");
+
+            Bot resource = CreateDefaultBotResource();
+
+            if (!string.IsNullOrEmpty(kind))
+            {
+                resource.Kind = kind;
+            }
+            
+            var responseResource =
+                botServiceMgmtClient.Bots.CreateWithHttpMessagesAsync(rgname, botid, resource).GetAwaiter().GetResult().Body;
+
+
+            VerifyBotServiceProperties(resource, responseResource);
+
+            return responseResource;
+        }
+
+        public static Bot CreateAndValidateWebBot(AzureBotServiceClient botServiceMgmtClient, string rgname,
+            string kind = null)
+        {
+            string botid = TestUtilities.GenerateName("botservice");
+
+            Bot resource = CreateDefaultBotResource();
+
+            if (string.IsNullOrEmpty(kind))
+            {
+                resource.Kind = Kind.Function;
+            }
+
+            BotDeploymentInfo deployInfo = new BotDeploymentInfo();
+            Bot responseResource = null;
+
+            if (resource.Kind == Kind.Function)
+            {
+                responseResource = botServiceMgmtClient.Bots.CreateFunctionBot(rgname, botid, resource, deployInfo);
+                VerifyBotServiceProperties(resource, responseResource, isRegistrationBot: false);
+            }
+            else
+            {
+                responseResource = botServiceMgmtClient.Bots.CreateWebAppBot(rgname, botid, resource, deployInfo);
+                VerifyBotServiceProperties(resource, responseResource, isRegistrationBot: false);
+            }
+            
+            return responseResource;
+        }
+
+        public static void VerifyBotServiceProperties(Bot expected, Bot actual, bool isCreation = false, bool isRegistrationBot = true)
```

Comment:
I'm guessing this is a test, probably should have the `[Fact]` attribute

Reply:
No, it is a helper method used by all tests. Instead of copying this code in all tests, we call this common method to avoid code duplication.